localstripe/resources.py

```diff
@@ -318,6 +318,22 @@ def _attaching_is_declined(self):
     def _charging_is_declined(self):
         return PaymentMethod._charging_is_declined(self)
 
+    @classmethod
+    def _api_list_all(cls, url, customer=None, limit=10):
+        try:
+            if customer is not None:
+                assert type(customer) is str and customer.startswith('cus_')
+        except AssertionError:
+            raise UserError(400, 'Bad request')
+
+        if customer:
+            Customer._api_retrieve(customer)  # to return 404 if not existant
+
+        li = super(Card, cls)._api_list_all(url, limit=limit)
```

Comment:
I think `super()` is enough.

Anyway, if you move this code inside the `Customer` class, you can simply call `Card._api_list_all()` which is even clearer :wink: